templates/base/server/db/__init__.py

```
@@ -0,0 +1,83 @@
+async def check_cache():
+    global cache 
+
+    if not cache:
+        raise CacheHealthCheckFailedException("Redis client is not initialized.")
+
+    try:
+        result = await cache.ping()
+        if not result:
+            raise CacheHealthCheckFailedException("Redis did not respond with PONG.")
+    except aioredis.RedisError as e:
+        raise CacheHealthCheckFailedException(f"Redis error: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)